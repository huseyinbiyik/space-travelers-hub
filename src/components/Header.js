import React from 'react';
import './styles/Header.css';
import { NavLink } from 'react-router-dom';
import logo from './images/logo.png';

export default function header() {
  const activeStyle = {
    textDecoration: 'underline',
  };

  return (
    <header className="App-header">

      <div className="main-wrapper">
        <div className="logo-wrapper">
          <img src={logo} className="App-logo" alt="logo" />
          <p className="logo-text">
            Space Traveler`s Hub
          </p>
        </div>

        <nav className="App-nav">
          <ul className="App-link-container">
            <li>
              <NavLink
                className="navbar-item"
                to="/"
                style={({ isActive }) => (isActive ? activeStyle : undefined)}
              >
                Rockets
              </NavLink>
            </li>
            <li>
              <NavLink
                className="navbar-item"
                to="/missions"
                style={({ isActive }) => (isActive ? activeStyle : undefined)}
              >
                Missions
              </NavLink>
            </li>
            <li>
              <NavLink
                className="navbar-item profile"
                to="/profile"
                style={({ isActive }) => (isActive ? activeStyle : undefined)}
              >
                My Profile
              </NavLink>
            </li>
          </ul>
        </nav>
      </div>

    </header>

  );
}
