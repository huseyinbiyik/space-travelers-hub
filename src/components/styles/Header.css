header {
  width: 100vw;
  height: 70px;
  border-bottom: 1px solid gainsboro;
  display: flex;
  align-items: center;
  margin-bottom: 20px;
}

/* We can use this as our main wrapper */
.main-wrapper {
  width: calc(100% - 200px);
  margin-left: 100px;
  margin-right: 100px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.logo-wrapper {
  display: flex;
  align-items: center;
  gap: 20px;
}

.App-logo {
  width: 36px;
  height: 36px;
}

.logo-text {
  font-size: 24px;
}

.App-nav {
  display: flex;
}

.App-link-container {
  display: flex;
  list-style-type: none;
}

.navbar-item {
  text-decoration: none;
  color: #017bfe;
  margin-left: 20px;
  margin-right: 20px;
}

.profile {
  margin-right: 0;
}

.App-link-container > li:last-child::before {
  content: '|';
}
