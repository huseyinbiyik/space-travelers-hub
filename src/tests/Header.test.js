import React from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';
import Header from '../components/Header';
import store from './../redux/configureStore.js';

describe('Header ui and snapshot test', () => {
  it('should render Header component', () => {
    const header = render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    );
    expect(header).toMatchSnapshot();
  });

  test('it should check if there is any element called Rockets on the header', () => {
    render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    );
    expect(screen.getByText('Rockets')).toBeInTheDocument();
  });
});
