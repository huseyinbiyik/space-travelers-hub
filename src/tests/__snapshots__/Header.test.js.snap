// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header ui and snapshot test should render Header component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="App-header"
      >
        <div
          class="main-wrapper"
        >
          <div
            class="logo-wrapper"
          >
            <img
              alt="logo"
              class="App-logo"
              src="logo.png"
            />
            <p
              class="logo-text"
            >
              Space Traveler\`s Hub
            </p>
          </div>
          <nav
            class="App-nav"
          >
            <ul
              class="App-link-container"
            >
              <li>
                <a
                  aria-current="page"
                  class="navbar-item active"
                  href="#/"
                  style="text-decoration: underline;"
                >
                  Rockets
                </a>
              </li>
              <li>
                <a
                  class="navbar-item"
                  href="#/missions"
                >
                  Missions
                </a>
              </li>
              <li>
                <a
                  class="navbar-item profile"
                  href="#/profile"
                >
                  My Profile
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="App-header"
    >
      <div
        class="main-wrapper"
      >
        <div
          class="logo-wrapper"
        >
          <img
            alt="logo"
            class="App-logo"
            src="logo.png"
          />
          <p
            class="logo-text"
          >
            Space Traveler\`s Hub
          </p>
        </div>
        <nav
          class="App-nav"
        >
          <ul
            class="App-link-container"
          >
            <li>
              <a
                aria-current="page"
                class="navbar-item active"
                href="#/"
                style="text-decoration: underline;"
              >
                Rockets
              </a>
            </li>
            <li>
              <a
                class="navbar-item"
                href="#/missions"
              >
                Missions
              </a>
            </li>
            <li>
              <a
                class="navbar-item profile"
                href="#/profile"
              >
                My Profile
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
